### ISSUE TITLE
unreachable!

### ISSUE THREAD

**Issue by malik672** [alloy-rs/alloy#60]
### Component

providers

### Describe the feature you would like

in the crate dir, under rpc-client/call.rs for error handling  the code currently uses unreachable! when it encounters an incorrect state. won't it be better to use custom errors Instead of using unreachable! or maybe i'm missing something

**Comment by malik672** [comment]
i know though, i'm talking about the error handling itself, won't it just be better to return an error ti the caller in case the incorrect state ever get's invoked

**Comment by prestwich** [comment]
no, the `unreachable!()` is not handling (or failing to handle) an error, it's communicating to the compiler that the program is incorrect if that line is ever reached

**Comment by malik672** [comment]
oh, thanks
